sw/device/lib/testing/sram_ctrl_testutils.c

```diff
@@ -10,84 +10,26 @@
 #include "sw/device/lib/runtime/log.h"
 #include "sw/device/lib/testing/check.h"
 
-/**
- * Main RAM parameters.
- */
-static const sram_ctrl_params_t sram_main = {
-    .reg_base = TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR,
-    .ram_start = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR,
-    .ram_end = TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR +
-               TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_SIZE_BYTES -
-               SRAM_CTRL_DATA_NUM_BYTES,
-};
-
-/**
- * Retention RAM parameters.
- */
-static const sram_ctrl_params_t sram_ret = {
-    .reg_base = TOP_EARLGREY_SRAM_CTRL_RET_AON_REGS_BASE_ADDR,
-    .ram_start = TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR,
-    .ram_end = TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR +
-               TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_SIZE_BYTES -
-               SRAM_CTRL_DATA_NUM_BYTES,
-};
-
-/**
- * Creates SRAM Control handle and initializes the peripheral.
- */
-static sram_ctrl_t sram_ctrl_init(const sram_ctrl_params_t *params) {
-  sram_ctrl_t sram = {
-      .params = params,
-  };
-
-  mmio_region_t region = mmio_region_from_addr(params->reg_base);
-  CHECK_DIF_OK(dif_sram_ctrl_init(region, &sram.dif));
-
-  return sram;
-}
-
-sram_ctrl_t sram_ctrl_main_init(void) { return sram_ctrl_init(&sram_main); }
-sram_ctrl_t sram_ctrl_ret_init(void) { return sram_ctrl_init(&sram_ret); }
-
-/**
- * Writes `data` at the `address` in RAM.
- */
-static void write_to_ram(uintptr_t address, const sram_ctrl_data_t *data) {
+void sram_ctrl_testutils_write(uintptr_t address,
+                               const sram_ctrl_data_t *data) {
   mmio_region_t region = mmio_region_from_addr(address);
   for (size_t index = 0; index < SRAM_CTRL_DATA_NUM_WORDS; ++index) {
```

Comment:
It will be useful to allow the test to decide what size payload to read / write to the SRAM. Suggest making `size` as a field of `sram_ctrl_data_t` (or something similar).

Reply:
We may also want to add OOB checks to catch bad arg errors sooner.

Reply:
```

Reply:
> We may also want to add OOB checks to catch bad arg errors sooner.

Are these the checks you would like me to add?

Reply:
I think it makes sense to make both as follow-up PRs and merge this in, as long as you are happy with the rest?

Reply:
I think it is OK to address @sriyerg's comments as a follow up PR.

Reply:
Added to #8546